pkg/kritis/cryptolib/verifier.go

```diff
@@ -49,13 +52,47 @@ type PublicKey struct {
 // NewPublicKey creates a new PublicKey. `keyType` contains the type of the
 // public key, one of Pgp, Pkix or Jwt. `keyData` contains the raw key
 // material. `keyID` contains a unique identifier for the public key. For PGP,
-// this should be the OpenPGP RFC4880 V4 fingerprint of the key.
-func NewPublicKey(keyType KeyType, keyData []byte, keyID string) PublicKey {
-	return PublicKey{
+// this should be the OpenPGP RFC4880 V4 fingerprint of the key. For PKIX and
+// JWT, the ID should contain valid URI characters.
+func NewPublicKey(keyType KeyType, keyData []byte, keyID string) (*PublicKey, error) {
+	switch keyType {
+	case Pgp:
+		err := validatePgpKeyID(keyData, keyID)
+		if err != nil {
+			return nil, err
+		}
+	case Pkix, Jwt:
+		// Valid URI characters (see http://tools.ietf.org/html/rfc3986#section-2)
+		reURI := regexp.MustCompile(`^[a-zA-Z0-9-._~:\/?#\[\]@!$&'\(\)*+,;=%]+$`)
```

Comment:
This seems like a reasonable restriction, but make sure it's documented in the public protos.
From discussions with Patrick, we have been thinking of Key Ids as a StringOrUri (https://tools.ietf.org/html/rfc7519) but I don't think we have formally defined them as such.

Reply:
Switched to StringOrUri.